webui/react/src/types.ts

```diff
@@ -237,22 +238,49 @@ export interface MetricName {
 
 // Checkpoint sub step.
 export interface Checkpoint extends StartEndTimes {
-  id: number;
   resources?: Record<string, number>;
   state: CheckpointState;
-  stepId: number;
   trialId: number;
   uuid? : string;
   validationMetric? : number;
 }
 
+export interface Workload extends StartEndTimes {
+  numBatches: number;
+  priorBatchesProcessed: number;
+}
+
+export interface CheckpointWorkload extends Workload {
+  resources?: Record<string, number>;
+  state: CheckpointState;
+  uuid? : string;
+}
+
+export interface MetricsWorkload extends Workload {
+  state: RunState;
+  metrics?: Record<string, number>;
+  numInputs?: number;
+}
+export interface WorkloadWrapper {
+  training?: MetricsWorkload;
+  validation?: MetricsWorkload;
+  checkpoint?: CheckpointWorkload;
+}
+
 // Validation sub step.
 export interface Validation extends StartEndTimes {
   id: number;
   state: RunState;
   metrics?: ValidationMetrics;
 }
 
+export interface Step2 extends StartEndTimes {
```

Comment:
**question:** not sure if this is any better:
```
export interface Step2 extends WorkloadWrapper, StartEndTimes {
```

Reply:
aah I thought TS wouldn't be happy with overwriting the `training` attribute but it didn't complain. this is better